sci-geosciences/gmt/gmt-5.4.4.ebuild

```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit bash-completion-r1 cmake-utils
+
+DESCRIPTION="Powerful map generator"
+HOMEPAGE="https://gmt.soest.hawaii.edu/"
+SRC_URI="mirror://gmt/${P}-src.tar.xz"
+
+LICENSE="GPL-2 gmttria? ( Artistic )"
+SLOT="5"
+KEYWORDS="~amd64 ~x86"
+IUSE="+dcw doc examples +fftw +gdal gmttria +gshhg htmldoc lapack openmp pcre pcre2 threads tutorial"
+
+RDEPEND="
+	!sci-biology/probcons
+	app-text/ghostscript-gpl
+	dcw? ( sci-geosciences/dcw-gmt-data )
+	fftw? ( sci-libs/fftw:3.0/3 )
+	gdal? ( sci-libs/gdal )
+	gshhg? ( sci-geosciences/gshhg-data )
+	>=sci-libs/netcdf-4.1[hdf5]
+	lapack? ( virtual/lapack )
+	pcre? ( dev-libs/libpcre )
+	pcre2? ( dev-libs/libpcre2 )
+	net-misc/curl
+	sys-libs/zlib"
+
+DEPEND="${RDEPEND}"
+
+REQUIRED_USE="
+	pcre? ( !pcre2 )
+	pcre2? ( !pcre )"
+
+src_prepare() {
+	cmake-utils_src_prepare
+	# Rename man pages to avoid a name conflict with gmt4
+	pushd man_release || die
+	local m c suffix newc
+	for m in *.gz; do
+		c=${m%%.*}
+		suffix=${m#*.}
+		# This man pages does'nt conflict
+		if [ "$c" == "gmt" ]; then continue; fi
+		if [ "$c" == "gmt.conf" ]; then continue; fi
+		if [ "$c" == "postscriptlight" ]; then continue; fi
+		newc=gmt_${c}
+		if [ "$c" == "gmt_shell_functions" ]; then newc=gmt5_shell_functions; fi
+		if [ "$c" == "gmtcolors" ]; then newc=gmt5colors; fi
+		mv "$c.$suffix" "$newc.$suffix" || die
+	done
+	popd || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DLICENSE_RESTRICTED="$(usex gmttria no yes)"
+		-DGMT_OPENMP="$(usex openmp)"
+		-DGMT_USE_THREADS="$(usex threads)"
+		-DGMT_INSTALL_TRADITIONAL_FOLDERNAMES=OFF # Install bash completions properly
+		-DGMT_INSTALL_MODULE_LINKS=OFF # Don't install symlinks on gmt binary, they are conflicted with gmt4
+		-DBASH_COMPLETION_DIR="$(get_bashcompdir)"
+		-DGMT_DATADIR="share/${P}"
+		-DGMT_DOCDIR="share/doc/${P}"
+		-DGMT_MANDIR="share/man"
+	)
+	use fftw || mycmakeargs+=( -DGMT_EXCLUDE_FFTW3=yes )
+	use gdal || mycmakeargs+=( -DGMT_EXCLUDE_GDAL=yes )
+	use lapack || mycmakeargs+=( -DGMT_EXCLUDE_LAPACK=yes )
+	use pcre || mycmakeargs+=( -DGMT_EXCLUDE_PCRE=yes )
+	use pcre2 || mycmakeargs+=( -DGMT_EXCLUDE_PCRE2=yes )
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+	# Remove various documentation
+	use doc || rm -rf "${ED}/usr/share/doc/${P}/pdf" || die
+	if use examples; then
+		docompress -x /usr/share/doc/${P}/examples
+	else
+		rm -rf "${ED}/usr/share/doc/${P}/examples" || die
+	fi
+	use htmldoc || rm -rf "${ED}/usr/share/doc/${P}/html" || die
+	if use tutorial; then
+		docompress -x /usr/share/doc/${P}/tutorial
+	else
+		rm -rf "${ED}/usr/share/doc/${P}/tutorial" || die
+	fi
+
+	# Rename some files to avoid a name conflict with gmt4
+	mv "${ED}/usr/bin/gmt_shell_functions.sh" "${ED}/usr/bin/gmt5_shell_functions.sh" || die
```

Comment:
Does this actually work? I'd guess the other parts still rely on the old name...

Reply:
No, these functions are for user use only. Never used them, however. Where the greater problem is the text of the manuals, which contain references to unchanged names.

The problem is that gmt4 and gmt5 have incompatibility with the commands options. As a result, I personally need to keep both versions in the system, since there is a bunch of old code that is hard to port. Perhaps it makes sense to create a new flag that will manage the compatibility of the two versions.